<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=notebooksDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=notebooksArray;
            var docs_map=notebooksArrayMap;
            var pos=notebooksArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Statistical Genetics Tutorials</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Statistical Genetics Tutorials</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../notebooks.html">Notebooks</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/statgenetics/statgen-courses"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[32]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># MR-JTI workflow</span>

<span class="c1">## Overview</span>

<span class="n">This</span> <span class="n">notebook</span> <span class="n">shows</span> <span class="n">the</span> <span class="n">workflow</span> <span class="k">for</span> <span class="n">MR</span><span class="o">-</span><span class="n">JTI</span><span class="p">,</span> <span class="n">which</span> <span class="n">utilizes</span> <span class="n">imputation</span> <span class="n">to</span> <span class="n">perform</span> <span class="n">trait</span> <span class="n">prediction</span> <span class="n">according</span> <span class="n">to</span> <span class="n">multi</span><span class="o">-</span><span class="n">tissue</span> <span class="n">relationship</span> <span class="n">gene</span> <span class="n">expression</span><span class="o">.</span> <span class="n">MR</span><span class="o">-</span><span class="n">JTI</span> <span class="n">acheives</span> <span class="n">a</span> <span class="n">higher</span> <span class="n">prediction</span> <span class="n">accuracy</span> <span class="n">by</span> <span class="n">leveraging</span> <span class="n">multi</span><span class="o">-</span><span class="n">tissue</span> <span class="n">information</span> <span class="ow">and</span> <span class="n">also</span> <span class="n">performes</span> <span class="n">Causal</span> <span class="n">Inference</span> <span class="n">on</span> <span class="n">trait</span> <span class="ow">and</span> <span class="n">gene</span> <span class="n">expression</span><span class="o">.</span>

 
<span class="sb">`Goal`</span>  
  
     <span class="mf">1.</span> <span class="n">Imputation</span> <span class="n">prediction</span> <span class="k">for</span> <span class="n">genes</span>
  
     <span class="mf">2.</span> <span class="n">Causal</span> <span class="n">Inference</span> <span class="n">between</span> <span class="n">Trait</span> <span class="ow">and</span> <span class="n">Gene</span> <span class="n">Expression</span>
     
  
<span class="sb">`Model`</span>
  
     <span class="n">Y</span> <span class="o">=</span> <span class="n">Trait</span>
  
     <span class="n">X</span> <span class="o">=</span> <span class="n">Predicted</span> <span class="n">expression</span> <span class="n">Gene</span> <span class="n">A</span>
  
     <span class="n">Data</span> <span class="o">=</span> <span class="n">transcriptome</span> <span class="n">data</span> <span class="k">as</span> <span class="n">the</span> <span class="n">information</span> <span class="n">about</span> <span class="n">the</span> <span class="n">mapping</span> <span class="n">between</span> <span class="n">Gene</span> <span class="n">A</span><span class="s1">&#39;s expression and SNP.</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h1 id="MR-JTI-workflow">MR-JTI workflow<a class="anchor-link" href="#MR-JTI-workflow">&#182;</a></h1><h2 id="Overview">Overview<a class="anchor-link" href="#Overview">&#182;</a></h2><p>This notebook shows the workflow for MR-JTI, which utilizes imputation to perform trait prediction according to multi-tissue relationship gene expression. MR-JTI acheives a higher prediction accuracy by leveraging multi-tissue information and also performes Causal Inference on trait and gene expression.</p>
<p><code>Goal</code></p>

<pre><code> 1. Imputation prediction for genes

 2. Causal Inference between Trait and Gene Expression


</code></pre>
<p><code>Model</code></p>

<pre><code> Y = Trait

 X = Predicted expression Gene A

 Data = transcriptome data as the information about the mapping between Gene A's expression and SNP.</code></pre>

</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[42]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">## Command Interface</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h2 id="Command-Interface">Command Interface<a class="anchor-link" href="#Command-Interface">&#182;</a></h2>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[16]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="o">%</span><span class="n">cd</span> <span class="s1">&#39;/Users/malong/Documents/neuro_twas&#39;</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>/Users/malong/Documents/neuro_twas</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[25]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">## MWE</span>

<span class="n">The</span> <span class="n">minimal</span> <span class="n">working</span> <span class="n">example</span> <span class="n">files</span> <span class="n">can</span> <span class="n">be</span> <span class="n">downloaded</span> <span class="kn">from</span> <span class="nn">dropbox</span> <span class="nn">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">dropbox</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">sh</span><span class="o">/</span><span class="n">i9elg3m4wav4o5g</span><span class="o">/</span><span class="n">AAABdxZbVyBclbfa_1KKVftDa</span><span class="err">?</span><span class="n">dl</span><span class="o">=</span><span class="mf">0.</span>
<span class="n">In</span> <span class="n">here</span> <span class="n">we</span> <span class="n">assume</span> <span class="n">the</span> <span class="n">folder</span> <span class="n">structure</span> <span class="ow">is</span> <span class="n">the</span> <span class="n">same</span> <span class="k">as</span> <span class="n">this</span> <span class="n">link</span><span class="o">.</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h2 id="MWE">MWE<a class="anchor-link" href="#MWE">&#182;</a></h2><p>The minimal working example files can be downloaded from dropbox <a href="https://www.dropbox.com/sh/i9elg3m4wav4o5g/AAABdxZbVyBclbfa_1KKVftDa?dl=0">https://www.dropbox.com/sh/i9elg3m4wav4o5g/AAABdxZbVyBclbfa_1KKVftDa?dl=0</a>.
In here we assume the folder structure is the same as this link.</p>

</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[26]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">nohup</span> <span class="n">sos</span> <span class="n">run</span> <span class="err">～</span><span class="o">/</span><span class="mo">0324</span><span class="n">_MR_JTI</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">JTI</span> \
  <span class="o">--</span><span class="n">tissue</span><span class="o">=</span><span class="n">Adipose_Subcutaneous</span> \
  <span class="o">--</span><span class="n">geneid</span><span class="o">=</span><span class="n">ENSG00000182957</span> \     <span class="c1"># for_each &#39;geneid&#39;</span>
  <span class="o">--</span><span class="n">genotype_path</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">jti_example_geno</span><span class="o">.</span><span class="n">bed</span> \
  <span class="o">--</span><span class="n">expression_path</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">jti_example_exp</span><span class="o">.</span><span class="n">txt</span> \
  <span class="o">--</span><span class="n">tmp_folder</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">tmp</span> \
  <span class="o">--</span><span class="n">gencode_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">gencode</span><span class="o">.</span><span class="n">v32</span><span class="o">.</span><span class="n">GRCh37</span><span class="o">.</span><span class="n">txt</span> \
  <span class="o">--</span><span class="n">plink_path</span> <span class="s2">&quot;/Users/Shared/OneDrive/Wang_Lab/plink_mac_20201019/plink&quot;</span>\
  <span class="o">--</span><span class="n">out_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span>     <span class="c1">#--outpath ${_output}</span>
 <span class="o">&amp;</span>

 
<span class="n">nohub</span> <span class="n">sos</span> <span class="n">docker</span> <span class="n">run</span> <span class="o">-</span><span class="n">it</span> <span class="o">--</span><span class="n">rm</span> <span class="o">--</span><span class="kp">name</span> <span class="n">my</span><span class="o">-</span><span class="n">running</span><span class="o">-</span><span class="n">script</span> <span class="o">-</span><span class="n">v</span> <span class="s2">&quot;$PWD&quot;</span><span class="p">:</span><span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">src</span><span class="o">/</span><span class="n">myapp</span> <span class="o">-</span><span class="n">w</span> <span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">src</span><span class="o">/</span><span class="n">myapp</span> <span class="n">python</span><span class="p">:</span><span class="mi">2</span> <span class="n">python</span> <span class="o">/</span><span class="n">Users</span><span class="o">/</span><span class="n">malong</span><span class="o">/</span><span class="n">Vpy27</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">MetaXcan</span><span class="o">.</span><span class="n">py</span>
  <span class="o">--</span><span class="n">model_db_path</span> <span class="err">$</span><span class="p">{</span><span class="n">input_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">model</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span><span class="o">.</span><span class="n">db</span> \
  <span class="o">--</span><span class="n">covariance_directory</span> <span class="err">$</span><span class="p">{</span><span class="n">input_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">model</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> \
  <span class="o">--</span><span class="n">gwas_file</span> <span class="err">$</span><span class="p">{</span><span class="n">input_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">trait</span><span class="p">}</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> \
  <span class="o">--</span><span class="n">snp_column</span> <span class="n">rsid</span> \
  <span class="o">--</span><span class="n">effect_allele_column</span> <span class="n">eff_allele</span> \
  <span class="o">--</span><span class="n">non_effect_allele_column</span> <span class="n">ref_allele</span> \
  <span class="o">--</span><span class="n">beta_column</span> <span class="n">beta</span> \
  <span class="o">--</span><span class="n">se_column</span> <span class="n">se</span> \
  <span class="o">--</span><span class="n">output_file</span> <span class="err">$</span><span class="p">{</span><span class="n">output_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">trait</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">model</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span><span class="o">.</span><span class="n">csv</span>


<span class="n">nohub</span> <span class="n">sos</span> <span class="n">run</span> <span class="o">~/</span><span class="mo">0324</span><span class="n">_MR_JTI</span><span class="o">.</span><span class="n">ipynb</span> <span class="n">MR</span> \
  <span class="o">--</span><span class="n">df_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">mrjti_example</span><span class="o">.</span><span class="n">txt</span> \
  <span class="o">--</span><span class="n">n_genes</span> <span class="mi">100</span> \
  <span class="o">--</span><span class="n">out_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">mrjti_result</span><span class="o">.</span><span class="n">csv</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_text output_error">
<pre>
<span class="ansi-red-intense-fg">File contains parsing errors: 
	[line  2]: nohup sos run ～/0324_MR_JTI.ipynb JTI \
  --tissue=Adipose_Subcutaneous \
  --geneid=ENSG00000182957 \     # for_each &#39;geneid&#39;
  --genotype_path=${example_file_dir}/jti_example_geno.bed \
  --expression_path=${example_file_dir}/jti_example_exp.txt \

Invalid statements: SyntaxError(&#39;invalid syntax&#39;, (&#39;&lt;string&gt;&#39;, 1, 7, &#39;nohup sos run ～/0324_MR_JTI.ipynb JTI \\\n&#39;))</span></pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[36]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">## Global Parameter Setting</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h2 id="Global-Parameter-Setting">Global Parameter Setting<a class="anchor-link" href="#Global-Parameter-Setting">&#182;</a></h2>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[17]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="c1"># path to the working directory</span>
<span class="kn">parameter:</span> <span class="n">wd</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># path to script where you put software JTI script</span>
<span class="kn">parameter:</span> <span class="n">script_dir</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/src&#39;</span>
<span class="c1"># path to exmpale file </span>
<span class="kn">parameter:</span> <span class="n">example_file_dir</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/data&#39;</span>
<span class="c1"># path to JTI python script</span>
<span class="kn">parameter:</span> <span class="n">py_script_dir</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/src/MetaXcan&#39;</span>
<span class="c1"># path to output file</span>
<span class="kn">parameter:</span> <span class="n">out_path</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/result&#39;</span>
<span class="c1"># path to your local PLINK software</span>
<span class="kn">parameter:</span> <span class="n">plink_path</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># target tissue name</span>
<span class="kn">parameter:</span> <span class="n">tissue</span> <span class="o">=</span> <span class="s2">&quot;chr&quot;</span>
<span class="c1"># provide the geneid which will be used to locate chromosome position</span>
<span class="kn">parameter:</span> <span class="n">geneid</span> <span class="o">=</span> <span class="s2">&quot;chr&quot;</span>
<span class="c1"># genotype file in PLINK format </span>
<span class="kn">parameter:</span> <span class="n">genotype_path</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/jti_example_geno.bed&#39;</span>
<span class="c1"># expression file</span>
<span class="kn">parameter:</span> <span class="n">expression_path</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/jti_example_exp.txt&#39;</span>
<span class="c1"># path for temperary files, will be clened up after model training</span>
<span class="kn">parameter:</span> <span class="n">tmp_folder</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/tmp&#39;</span>
<span class="c1"># the gene annotation file</span>
<span class="kn">parameter:</span> <span class="n">gencode_path</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/gencode.v32.GRCh37.txt&#39;</span>




<span class="c1"># Get geneid of interest to focus on.</span>
<span class="n">geneid</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">()</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">gencode_path</span><span class="p">)</span><span class="o">.</span><span class="n">readlines</span><span class="p">()</span> <span class="k">if</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;#&#39;</span><span class="p">)]</span>

<span class="c1"># NEEDFIX:</span>
<span class="c1"># molecular_pheno = [x.strip().split() for x in open(molecular_pheno_dir).readlines() if x.strip() and not x.strip().startswith(&#39;#&#39;)]</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[29]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">## JTI Input</span>

<span class="o">*</span> <span class="o">--</span><span class="n">tissue</span><span class="p">,</span> <span class="n">target</span> <span class="n">tissue</span> <span class="kp">name</span> 

<span class="o">*</span> <span class="o">--</span><span class="n">geneid</span><span class="p">,</span> <span class="n">gene</span> <span class="nb">id</span> <span class="n">The</span> <span class="n">ENSG</span> <span class="n">gene</span> <span class="n">ID</span><span class="o">.</span> <span class="n">Provide</span> <span class="n">the</span> <span class="n">real</span> <span class="n">ENSG</span> <span class="n">gene</span> <span class="n">ID</span> <span class="n">which</span> <span class="n">will</span> <span class="n">be</span> <span class="n">used</span> <span class="n">to</span> <span class="n">find</span> <span class="n">the</span> <span class="n">chromosome</span> <span class="ow">and</span> <span class="n">position</span> <span class="k">for</span> <span class="n">the</span> <span class="n">gene</span><span class="o">.</span>

<span class="o">*</span> <span class="o">--</span><span class="n">genotype_path</span><span class="p">,</span> <span class="n">Genotype</span> <span class="nb">file</span> <span class="ow">in</span> <span class="n">plink</span> <span class="n">bfile</span> <span class="n">format</span> <span class="s1">&#39;.bed, .bim, .fam&#39;</span><span class="p">,</span> <span class="n">used</span> <span class="k">for</span> <span class="n">data</span> <span class="n">preprocessing</span> <span class="ow">in</span> <span class="n">QC</span><span class="p">,</span> <span class="n">filtering</span><span class="p">,</span> <span class="n">calculating</span> <span class="n">MAF</span><span class="p">,</span> <span class="n">call</span> <span class="n">rate</span><span class="p">,</span> <span class="ow">and</span> <span class="n">R</span><span class="o">-</span><span class="n">sq</span><span class="p">,</span>
<span class="n">it</span> <span class="n">contains</span> <span class="n">a</span> <span class="n">snp</span> <span class="n">matrix</span> <span class="k">for</span> <span class="n">snp</span> <span class="kp">name</span> <span class="ow">and</span> <span class="n">family</span> <span class="n">information</span><span class="o">.</span> <span class="n">The</span> <span class="n">example</span> <span class="nb">file</span> <span class="n">here</span> <span class="ow">is</span> <span class="n">like</span><span class="p">:</span> <span class="err">$</span><span class="p">{</span><span class="n">genotype_path</span><span class="p">}</span><span class="o">.</span><span class="n">bed</span><span class="o">/</span><span class="n">fam</span><span class="o">/</span><span class="n">bim</span>

      <span class="n">jti_example_geno</span><span class="o">.</span><span class="n">bed</span>
      <span class="n">jti_example_geno</span><span class="o">.</span><span class="n">bim</span>
      <span class="n">jti_example_geno</span><span class="o">.</span><span class="n">fam</span>

<span class="o">*</span> <span class="o">--</span><span class="n">expression_path</span><span class="p">,</span> <span class="n">expression</span> <span class="n">data</span> <span class="n">that</span> <span class="n">will</span> <span class="n">be</span> <span class="n">used</span> <span class="n">to</span> <span class="n">normalize</span> <span class="ow">and</span> <span class="n">residulize</span> <span class="k">for</span> <span class="n">age</span><span class="p">,</span> <span class="n">gender</span><span class="p">,</span> <span class="n">PCs</span><span class="p">,</span> <span class="n">PEERs</span><span class="p">,</span> <span class="ow">and</span> <span class="n">so</span> <span class="n">on</span><span class="o">.</span> <span class="n">It</span> <span class="n">contains</span> <span class="n">tissue</span> <span class="kp">name</span><span class="p">,</span> <span class="n">sample</span> <span class="nb">id</span> <span class="ow">and</span> <span class="n">expression</span> <span class="n">level</span>

<span class="o">*</span> <span class="o">--</span><span class="n">gencode_path</span><span class="p">,</span> <span class="n">gene</span> <span class="n">annotation</span> <span class="nb">file</span> <span class="ow">in</span> <span class="o">.</span><span class="n">txt</span> <span class="n">format</span><span class="p">,</span> <span class="n">includes</span> <span class="n">each</span> <span class="n">gene</span><span class="s1">&#39;s gene id, name, strand and so on. We use the colum &#39;</span><span class="n">geneid</span><span class="s1">&#39; as the list to iterate our workflow through all the genes.</span>

<span class="o">*</span> <span class="o">--</span><span class="n">plink_path</span>

       <span class="n">Path</span> <span class="n">to</span> <span class="n">plink</span> <span class="n">software</span><span class="o">.</span> <span class="n">If</span> <span class="ow">not</span> <span class="n">specified</span><span class="p">,</span> <span class="n">you</span> <span class="n">will</span> <span class="n">need</span> <span class="n">to</span> <span class="n">add</span> <span class="n">the</span> <span class="n">path</span> <span class="n">of</span> <span class="n">plink</span> <span class="n">to</span> <span class="err">$</span><span class="n">PATH</span> <span class="ow">in</span> <span class="o">~/.</span><span class="n">bash_profile</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h2 id="JTI-Input">JTI Input<a class="anchor-link" href="#JTI-Input">&#182;</a></h2><ul>
<li><p>--tissue, target tissue name</p>
</li>
<li><p>--geneid, gene id The ENSG gene ID. Provide the real ENSG gene ID which will be used to find the chromosome and position for the gene.</p>
</li>
<li><p>--genotype_path, Genotype file in plink bfile format '.bed, .bim, .fam', used for data preprocessing in QC, filtering, calculating MAF, call rate, and R-sq,
it contains a snp matrix for snp name and family information. The example file here is like: ${genotype_path}.bed/fam/bim</p>

<pre><code>jti_example_geno.bed
jti_example_geno.bim
jti_example_geno.fam</code></pre>
</li>
<li><p>--expression_path, expression data that will be used to normalize and residulize for age, gender, PCs, PEERs, and so on. It contains tissue name, sample id and expression level</p>
</li>
<li><p>--gencode_path, gene annotation file in .txt format, includes each gene's gene id, name, strand and so on. We use the colum 'geneid' as the list to iterate our workflow through all the genes.</p>
</li>
<li><p>--plink_path</p>

<pre><code> Path to plink software. If not specified, you will need to add the path of plink to $PATH in ~/.bash_profile</code></pre>
</li>
</ul>

</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="n">JTI</span><span class="p">]</span>
<span class="kn">input:</span> <span class="n">example_file_dir</span><span class="p">,</span> <span class="n">for_each</span> <span class="o">=</span> <span class="s2">&quot;geneid&quot;</span>
<span class="kn">output:</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/{_geneid[0]}_{tissue}.txt&#39;</span>   <span class="c1">#ENSG00000182957_Adipose_Subcutaneous.txt</span>

<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;{ }&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[1]}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[1]}.stdout&#39;</span><span class="p">,</span><span class="n">container</span> <span class="o">=</span> <span class="n">container</span>

<span class="c1">#-----start-----</span>
<span class="c1"># ml GCC OpenMPI R</span>
<span class="c1"># openMPI : multi-progress tool, on CU Cluster</span>

    <span class="n">Rscript</span> <span class="err">$</span><span class="p">{</span><span class="n">script_dir</span><span class="p">}</span><span class="o">/</span><span class="n">JTI_yutian_debuged</span><span class="o">.</span><span class="n">r</span> \
      <span class="o">--</span><span class="n">tissue</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">geneid</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">geneid</span><span class="p">}</span> \              <span class="c1"># for_each &#39;geneid&#39;</span>
      <span class="o">--</span><span class="n">genotype_path</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">jti_example_geno</span><span class="o">.</span><span class="n">bed</span> \
      <span class="o">--</span><span class="n">expression_path</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">jti_example_exp</span><span class="o">.</span><span class="n">txt</span> \
      <span class="o">--</span><span class="n">tmp_folder</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">tmp_folder</span><span class="p">}</span> \
      <span class="o">--</span><span class="n">gencode_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">gencode</span><span class="o">.</span><span class="n">v32</span><span class="o">.</span><span class="n">GRCh37</span><span class="o">.</span><span class="n">txt</span> \
      <span class="o">--</span><span class="n">plink_path</span> <span class="s2">&quot;${plink_path}/plink&quot;</span>\
      <span class="o">--</span><span class="n">out_path</span> <span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">:</span><span class="n">b</span><span class="p">}</span>        <span class="c1"># :b option: make output file to ${_input[0]}, ${_input[1]} for next step</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[34]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">## AssoTest Input</span>

<span class="o">*</span> <span class="o">--</span><span class="n">db_path</span><span class="p">,</span> <span class="n">prediciton</span> <span class="n">model</span><span class="p">,</span> <span class="n">pre</span><span class="o">-</span><span class="n">trained</span> <span class="ow">and</span> <span class="ow">in</span> <span class="n">db</span> <span class="n">format</span>

<span class="o">*</span> <span class="o">--</span><span class="n">snp_cov</span><span class="p">,</span> <span class="n">snp</span> <span class="n">covariance</span> <span class="n">matrix</span> <span class="n">estimated</span> <span class="kn">from</span> <span class="nn">a</span> <span class="nn">reference</span> <span class="nn">dataset</span> <span class="p">(</span><span class="n">e</span><span class="o">.</span><span class="n">g</span><span class="o">.</span><span class="p">,</span> <span class="mi">1000</span><span class="n">g</span><span class="p">,</span> <span class="n">GTEx</span><span class="p">)</span><span class="o">.</span> <span class="n">The</span> <span class="n">covariance</span> <span class="n">matrix</span> <span class="ow">is</span> <span class="n">needed</span> <span class="k">for</span> <span class="n">association</span> <span class="n">test</span> <span class="n">using</span> <span class="n">GWAS</span> <span class="n">summary</span> <span class="n">statistics</span><span class="o">.</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h2 id="AssoTest-Input">AssoTest Input<a class="anchor-link" href="#AssoTest-Input">&#182;</a></h2><ul>
<li><p>--db_path, prediciton model, pre-trained and in db format</p>
</li>
<li><p>--snp_cov, snp covariance matrix estimated from a reference dataset (e.g., 1000g, GTEx). The covariance matrix is needed for association test using GWAS summary statistics.</p>
</li>
</ul>

</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="n">AssoTest</span><span class="p">]</span>
<span class="kn">input:</span> <span class="n">db_path</span>
<span class="kn">output:</span> <span class="s1">&#39;${trait}_${model}_${tissue}.csv&#39;</span>

<span class="c1"># a pre-downloaded database with prediction models</span>
<span class="kn">parameter:</span> <span class="n">db_path</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># snp covariance matrix</span>
<span class="kn">parameter:</span> <span class="n">snp_cov</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># model used in AssoTest</span>
<span class="kn">parameter:</span> <span class="n">model</span> <span class="o">=</span> <span class="s1">&#39;JTI&#39;</span>
<span class="c1"># data folder</span>
<span class="kn">parameter:</span> <span class="n">input_folder</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{wd:a}/data&#39;</span>

<span class="kn">R:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;$[ ]&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    <span class="c1">#ml GCC/8.2.0  OpenMPI/3.1.4 Intel/2019.1.144  IntelMPI/2018.4.274 R/3.6.0</span>
    <span class="n">library</span><span class="p">(</span><span class="n">RSQLite</span><span class="p">)</span>
    <span class="n">con</span> <span class="o">&lt;-</span> <span class="n">dbConnect</span><span class="p">(</span><span class="n">RSQLite</span><span class="p">::</span><span class="n">SQLite</span><span class="p">(),</span> <span class="n">dbname</span><span class="o">=</span><span class="s1">&#39;${db_path}&#39;</span><span class="p">)</span>             <span class="c1">#establish connections</span>
    <span class="n">dbListTables</span><span class="p">(</span><span class="n">con</span><span class="p">)</span>  <span class="c1">#datasets</span>
    <span class="n">dbListFields</span><span class="p">(</span><span class="n">con</span><span class="p">,</span> <span class="s1">&#39;weights&#39;</span><span class="p">)</span>   <span class="c1">#cols</span>
    <span class="n">weights</span> <span class="o">=</span> <span class="n">dbReadTable</span><span class="p">(</span><span class="n">con</span><span class="p">,</span><span class="s2">&quot;weights&quot;</span><span class="p">)</span>
    <span class="n">dbDisconnect</span><span class="p">(</span><span class="n">con</span><span class="p">)</span> <span class="c1">#disconnect</span>

<span class="c1"># convert to dosage file</span>
<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;$[ ]&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    <span class="n">plink</span> <span class="o">--</span><span class="n">bfile</span> <span class="o">%</span><span class="p">{</span><span class="n">genotype_path</span><span class="p">}</span> <span class="o">--</span><span class="n">recode</span> <span class="n">A</span> <span class="o">--</span><span class="n">out</span> <span class="o">./</span><span class="n">dosage</span>


<span class="c1">#association test</span>
    <span class="c1">#ml GCC/5.4.0-2.26  OpenMPI/1.10.3 pandas/0.18.1-Python-2.7.12 numpy/1.11.1-Python-2.7.12 scipy/0.17.0-Python-2.7.12</span>
    <span class="err">$</span><span class="p">{</span><span class="n">py_script_dir</span><span class="p">}</span><span class="o">/</span><span class="n">MetaXcan</span><span class="o">.</span><span class="n">py</span> 
        <span class="o">--</span><span class="n">model_db_path</span> <span class="err">$</span><span class="p">{</span><span class="n">input_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">model</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span><span class="o">.</span><span class="n">db</span> \
        <span class="o">--</span><span class="n">covariance_directory</span> <span class="err">$</span><span class="p">{</span><span class="n">input_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">model</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">gwas_file</span> <span class="err">$</span><span class="p">{</span><span class="n">input_folder</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">trait</span><span class="p">}</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">snp_column</span> <span class="n">rsid</span> \
        <span class="o">--</span><span class="n">effect_allele_column</span> <span class="n">eff_allele</span> \
        <span class="o">--</span><span class="n">non_effect_allele_column</span> <span class="n">ref_allele</span> \
        <span class="o">--</span><span class="n">beta_column</span> <span class="n">beta</span> \
        <span class="o">--</span><span class="n">se_column</span> <span class="n">se</span> \
        <span class="o">--</span><span class="n">output_file</span> <span class="err">$</span><span class="p">{</span><span class="n">out_path</span><span class="p">}</span><span class="o">/</span><span class="err">$</span><span class="p">{</span><span class="n">trait</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">model</span><span class="p">}</span><span class="n">_</span><span class="err">$</span><span class="p">{</span><span class="n">tissue</span><span class="p">}</span><span class="o">.</span><span class="n">csv</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[35]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">## MR Input</span>

<span class="o">*</span> <span class="o">--</span> <span class="n">mrjti_example</span><span class="p">,</span> <span class="n">The</span> <span class="n">SNP</span><span class="p">,</span> <span class="n">eQTL</span> <span class="ow">and</span> <span class="n">GWAS</span> <span class="n">estimation</span> <span class="n">information</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>



<div class="output_markdown rendered_html output_subarea ">
<h2 id="MR-Input">MR Input<a class="anchor-link" href="#MR-Input">&#182;</a></h2><ul>
<li>-- mrjti_example, The SNP, eQTL and GWAS estimation information</li>
</ul>

</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[22]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="n">MR</span><span class="p">]</span>
<span class="kn">input:</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">mrjti_example</span><span class="o">.</span><span class="n">txt</span>
<span class="kn">output:</span> <span class="s1">&#39;mrjti_result.csv&#39;</span>

<span class="kn">bash:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s2">&quot;$[ ]&quot;</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="n">f</span><span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    <span class="n">Rscript</span> <span class="err">$</span><span class="p">{</span><span class="n">script_dir</span><span class="p">}</span><span class="o">/</span><span class="n">MR</span><span class="o">-</span><span class="n">JTI</span><span class="o">.</span><span class="n">r</span> \
        <span class="o">--</span><span class="n">df_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">mrjti_example</span><span class="o">.</span><span class="n">txt</span> \
        <span class="o">--</span><span class="n">n_genes</span> <span class="mi">100</span> \
        <span class="o">--</span><span class="n">out_path</span> <span class="err">$</span><span class="p">{</span><span class="n">example_file_dir</span><span class="p">}</span><span class="o">/</span><span class="n">mrjti_result</span><span class="o">.</span><span class="n">csv</span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2019 Center for Statistical Genetics, Department of Neurology, Columbia University
<p><small>Exported from <a href="http://github.com/statgenetics/statgen-courses/blob/5aca3b9c0e14ce3ad374d1b854c15da9bedf3a14/notebooks/MR_JTI.ipynb"><code>notebooks/MR_JTI.ipynb</code></a> committed by nickluo1313 on Thu Mar 25 03:18:50 2021 <a href="http://github.com/statgenetics/statgen-courses/commit/5aca3b9c0e14ce3ad374d1b854c15da9bedf3a14">revision 2, 5aca3b9</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
